Google.Api.Generator/CodeGenerator.cs

```diff
@@ -116,15 +120,35 @@ internal static class CodeGenerator
                         $"Found namespaces '{string.Join(", ", namespaces)}' in package '{singlePackageFileDescs.Key}'.");
                 }
                 var catalog = new ProtoCatalog(singlePackageFileDescs.Key, descriptors, singlePackageFileDescs, commonResourcesConfigs);
-                foreach (var resultFile in GeneratePackage(namespaces[0], singlePackageFileDescs, catalog, clock, grpcServiceConfig))
+                foreach (var resultFile in GeneratePackage(namespaces[0], singlePackageFileDescs, catalog, clock, grpcServiceConfig, serviceConfig))
                 {
                     yield return resultFile;
                 }
             }
         }
 
+        private static Service ParseServiceConfigYaml(string path)
+        {
+            if (path is null)
+            {
+                return null;
+            }
+            
+            var deserializer = new Deserializer();
+            using (var reader = File.OpenText(path))
+            {
+                var yamlObject = deserializer.Deserialize(reader);
+                var serializer = new SerializerBuilder().JsonCompatible().Build();
+                var writer = new StringWriter();
+                serializer.Serialize(writer, yamlObject);
+                string json = writer.ToString();
+                var parser = new JsonParser(JsonParser.Settings.Default.WithIgnoreUnknownFields(true));
```

Comment:
Possibly - but it would be significantly more work. I don't find the YAML library terribly easy to use, whereas just converting to JSON and then using that JSON is much simpler :)

Reply:
I ended up going through json in my attempt as well, altho in my case I went for ` Service.Parser.ParseJson()`.